pkgs/development/python-modules/freud/default.nix

```diff
@@ -0,0 +1,63 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, cmake
+, cython
+, oldest-supported-numpy
+, scikit-build
+, setuptools
+, wheel
+, tbb
+, numpy
+, rowan
+, scipy
+}:
+
+buildPythonPackage rec {
+  pname = "freud";
+  version = "3.0.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "glotzerlab";
+    repo = "freud";
+    rev = "v${version}";
+    hash = "sha256-aKh2Gub1vU/wzvWkCl8yzlIswp8CtR975USiCr6ijUI=";
+    fetchSubmodules = true;
+  };
+  # Because we prefer to not `leaveDotGit`, we need to fool upstream into
+  # thinking we left the .git files in the submodules, so cmake won't think we
+  # didn't initialize them.
+  preConfigure = builtins.map (submodule: ''
+    touch extern/${submodule}/.git
+  '') [ "fsph" "Eigen" "voro++" ];
```

Comment:
Why not use our Eigen?

Reply:
> Why not use our Eigen?

Indeed it would have been nice, I haven't thought about it. However, upstream doesn't seem to support that out of the box via a cmake option or a like. Also I'm not sure it will save us space, because upstream doesn't distribute eventually all of eigen's libraries. I added a sentence to the comment explaining the reasoning.